File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example1/CommercialAircraft.cs
namespace MediatorDesignPattern.Example1;

/// <summary>
/// Concrete Colleague - Commercial passenger aircraft
/// </summary>
public class CommercialAircraft : Aircraft
{
    public CommercialAircraft(IMediator mediator, string callSign) 
        : base(mediator, callSign, "Commercial")
    {
    }

    public override void Send(string message)
    {
        Console.WriteLine($"[{CallSign}] Sending: {message}");
        Mediator.SendMessage(message, this);
    }

    public override void Receive(string message)
    {
        Console.WriteLine($"[{CallSign}] Received: {message}");
    }

    public override void RequestLanding()
    {
        Console.WriteLine($"[{CallSign}] Requesting landing clearance...");
        Mediator.RequestLanding(this);
    }

    public override void RequestTakeoff()
    {
        Console.WriteLine($"[{CallSign}] Requesting takeoff clearance...");
        Mediator.RequestTakeoff(this);
    }
}

File: java/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example1/src/main/java/behavioral/mediator/example1/CommercialAircraft.java
package behavioral.mediator.example1;

public class CommercialAircraft extends Aircraft {
    public CommercialAircraft(IMediator mediator, String callSign) {
        super(mediator, callSign, "Commercial");
    }

    @Override
    public void send(String message) { mediator.sendMessage(message, this); }

    @Override
    public void receive(String message) {
        System.out.println("[" + callSign + "] Received: " + message);
    }

    @Override
    public void requestLanding() { mediator.requestLanding(this); }

    @Override
    public void requestTakeoff() { mediator.requestTakeoff(this); }
}
